sdks/python/apache_beam/pipeline.py

```diff
@@ -509,19 +514,25 @@ def run(self, test_runner_api='AUTO'):
             self.runner.__class__.__name__ == 'SwitchingDirectRunner' and
             not self._options.view_as(StandardOptions).streaming)
 
+        # Multi-language pipelines that contain external pipeline segments may
+        # not be able to create a Python pipeline object graph. Hence following
+        # runner API check should be skipped for such pipelines.
+        external_transform_finder = ExternalTransformFinder()
+        self.visit(external_transform_finder)
```

Comment:
Don't we have to do something with external_transform_finder.contains_external_transforms? (Or is this fully redundant with the statement above?)

(IMHO, this visitor pattern is a bit error-prone, perhaps it'd be worth a static method on ExternalTransformFinder that does the visiting and returns a boolean.)

Reply:
This was redundant. Updated.

Also, updated the visitor class by adding a static method like you mentioned.